python/adelphi/adelphi/store.py

```diff
@@ -0,0 +1,87 @@
+# Functions to facilitate interactions with the underlying data store
+
+import logging
+from itertools import tee
+
+# Account for name change in itertools as of py3k
+try:
+    from itertools import ifilterfalse as filterfalse
+except ImportError:
+    from itertools import filterfalse
+
+from cassandra.cluster import Cluster, ExecutionProfile, EXEC_PROFILE_DEFAULT, default_lbp_factory
+from cassandra.auth import PlainTextAuthProvider
+
+logging.basicConfig(level=logging.INFO)
+log = logging.getLogger('adelphi')
+
+system_keyspaces = set(["system",
+                    "system_schema",
+                    "system_traces",
+                    "system_auth",
+                    "system_distributed",
+                    "system_virtual_schema",
+                    "system_views"])
+
+def build_auth_provider(username = None,password = None):
+    # instantiate auth provider if credentials have been provided
+    auth_provider = None
+    if username is not None and password is not None:
+        auth_provider = PlainTextAuthProvider(username=username, password=password)
+    return auth_provider
+
+
+def with_cluster(cluster_fn, hosts, port, username = None, password = None):
+    ep = ExecutionProfile(load_balancing_policy=default_lbp_factory())
+    cluster = Cluster(hosts, port=port, auth_provider=build_auth_provider(username,password), execution_profiles={EXEC_PROFILE_DEFAULT: ep})
+    log.info("Connecting to the cluster to get metadata...")
+    cluster.connect()
+    cluster_fn(cluster)
+    cluster.shutdown()
+
+
+def build_keyspace_objects(keyspaces, metadata):
```

Comment:
This seemed to be a better name for this function.  This guy isn't just about filtering keyspaces... he's also about getting a sequence of keyspace _objects_ for a given set of keyspace _names_.